pkg/controller/eventinstance/eventinstance.go

```diff
@@ -0,0 +1,91 @@
+package eventinstance
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	"github.com/acorn-io/acorn/pkg/config"
+	"github.com/acorn-io/baaah/pkg/router"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const defaultTTL = 168 * time.Hour // 7 days
+
+func GCExpired() router.HandlerFunc {
+	// last stores pre-parsed TTLs from the configuration.
+	last := new(struct {
+		mu  sync.Mutex
+		raw string
+		ttl time.Duration
+	})
+
+	return handler{
+		getTTL: func(ctx context.Context, getter kclient.Reader) (time.Duration, error) {
+			cfg, err := config.Get(ctx, getter)
+			if err != nil {
+				return 0, err
+			}
+
+			cfgTTL := cfg.EventTTL
+			if cfgTTL == nil || *cfgTTL == "" {
+				return defaultTTL, nil
+			}
+
+			last.mu.Lock()
+			defer last.mu.Unlock()
+
+			if last.raw != *cfgTTL {
+				// This is a new TTL, parse and memoize
+				ttl, err := time.ParseDuration(*cfgTTL)
+				if err != nil {
+					return 0, err
+				}
+
+				last.raw, last.ttl = *cfgTTL, ttl
+			}
+
+			return last.ttl, nil
+		},
+	}.gcExpired
+}
+
+// ttlFunc is a function that returns the TTL to use for event expiration.
+type ttlFunc func(context.Context, kclient.Reader) (time.Duration, error)
+
+type handler struct {
+	getTTL ttlFunc
+}
+
+func (h handler) gcExpired(req router.Request, resp router.Response) error {
+	e := req.Object.(*v1.EventInstance)
+
+	// Get the currently configured TTL
+	ttl, err := h.getTTL(req.Ctx, req.Client)
```

Comment:
What advantage is there to have this separate handler object with the `getTTL` function on it, and the `last` struct that it maintains? Is it just to prevent us from having to parse the TTL every time this function is called? Because currently it is still going to fetch the Acorn config every time this function is called.

Reply:
> Is it just to prevent us from having to parse the TTL every time this function is called? 

there's a few advantages to the approach:
- I can swap out the `getTTL` function in unit tests more easily
- Keeping the `handler` and `getTTL` unexported, prevents the "real" usage from getting to complicated
- don't need to parse TTL every time

> Because currently it is still going to fetch the Acorn config every time this function is called.

this is on purpose, because the config can change. but there's a discussion going on about this and I'm probably going to change this (see https://github.com/acorn-io/acorn/pull/1719#discussion_r1220052471)